lib/validate-options.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @fileoverview Validates rule options.
```

Comment:
I think this is named incorrectly. I'd call it config validator, as we may even want to add further validations, like for deprecated or removed rules.